### PR TITLE
Create a global CONTRIBUTING page for repositories which do not override it

### PR DESCRIPTION
**Pull request by oleg-nenashev**
First draft of the global Contributing page for those repositories which do not have it

### CODE REVIEW

**Comment by halkeye** [comment]
same file for infra too?

**Comment by oleg-nenashev** [comment]
> same file for infra too?

@halkeye Already there: https://github.com/jenkins-infra/.github/blob/master/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+[![Contributing](https://img.shields.io/badge/contributing-guidelines-blue)](https://jenkins.io/participate/)
+
+In the Jenkins project we appreciate any kind of contributions: code, documentation, design, etc.
+Any contribution counts, and the size does not matter!
+Check out [this page](https://jenkins.io/participate/) for more information and links!
+
+Many plugins and components also define their own contributing guidelines and communication channels. 
+There is also a big number of [mailing lists](https://jenkins.io/mailing-lists/) and [chats](https://jenkins.io/chat/).
+
+| WARNING: This is a default CONTRIBUTING page for all repositories in Jenkins. It might be overridden in repositories, and contributing guidelines maybe also defined in other places README. |
```
Comment (daniel-beck) [inline CONTRIBUTING.md]: Very important, but perhaps less scary? Explain that in a way, every plugin is its own subproject, and its maintainers may deviate in their rules, so if the repo has a README etc., it takes precedence over this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Contributing to Jenkins
+
+[![Contributing](https://img.shields.io/badge/contributing-guidelines-blue)](https://jenkins.io/participate/)
```
Comment (daniel-beck) [inline CONTRIBUTING.md]: But why?

Comment (oleg-nenashev) [reply CONTRIBUTING.md]: hm?

Comment (daniel-beck) [reply CONTRIBUTING.md]: I don't understand the use of badges as, apparently, decorative elements.

Comment (oleg-nenashev) [reply CONTRIBUTING.md]: It was rather a quick link for those who do not want to read text. Happy to revert it

```suggestion
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (oleg-nenashev) [reply CONTRIBUTING.md]:
```suggestion
| WARNING: This is a default CONTRIBUTING page for all repositories in Jenkins. Every plugin/component is its own sub-project which may deviate in its rules. Guidelines and channel links in repository CONTRIBUTING/README pages, if any, take precedence over this page. |
```

Comment (oleg-nenashev) [reply CONTRIBUTING.md]: @daniel-beck WDYT?

File: CONTRIBUTING.md (hunk 2 above)
Comment (daniel-beck) [reply CONTRIBUTING.md]: Sure, having a link makes sense, I'm just confused about the badge as link label. Why not write something like the following instead?

> Want to learn more about the various ways of contributing to Jenkins? See our guides on jenkins.io.

Seems more accessible that way (less ambiguous).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+In the Jenkins project we appreciate any kind of contributions: code, documentation, design, etc.
+Any contribution counts, and the size does not matter!
+Check out [this page](https://jenkins.io/participate/) for more information and links!
+
+Many plugins and components also define their own contributing guidelines and communication channels. 
+There is also a big number of [mailing lists](https://jenkins.io/mailing-lists/) and [chats](https://jenkins.io/chat/).
+
+| WARNING: This is a default CONTRIBUTING page for all repositories in Jenkins. Every plugin/component is its own sub-project which may deviate in its rules. Guidelines and channel links in repository CONTRIBUTING/README pages, if any, take precedence over this page. |
```
Comment (daniel-beck) [inline CONTRIBUTING.md]: The text is now a lot better, but presentation (bold, all-caps "WARNING") still looks scary -- like something that not reading would have _very_ negative consequences.

If this is intentional, that's fine. I don't think this warrants it, though.

Comment (oleg-nenashev) [reply CONTRIBUTING.md]:
```suggestion
| NOTE: This is a default CONTRIBUTING page for all repositories in Jenkins. Every plugin/component is its own sub-project which may deviate in its rules. Guidelines and channel links in repository CONTRIBUTING/README pages, if any, take precedence over this page. |
```

**Comment by oleg-nenashev** [comment]
Now it's a NOTE @daniel-beck

**Comment by daniel-beck** [comment]
I think even WARNING was fine -- I suspect it's the table formatting using a TH instead of TR, and it's bold by default?

Anyway, detail 🤷‍♂

**Pull request merged**